elotl/nahuatl/morphology.py

```diff
@@ -34,31 +25,19 @@ class Analyser(elotl.utils.morphology.Analyser):
 	tokeniser: function
 		A tokenisation function, if none is provided a default tokeniser, _tokenise()
 		is used which is based on regular expressions.
-
+	normalize: bool
+		Whether of not to perform orthographic normalization on the input prior to
+		passing it to the analyzer. Each language's analyzer should have its own
+		default target normalized orthography (depending on the orthogaphy used in the analyzer).
+		In some cases this may be beneficial. Often, though, it may not be necessary,
+		since most of the morphological analyzers, at least those that we currently leverage
+		from the apertium project, were built with a "spellrelax" component that enables
+		flexible input orthography.
 	"""
-	def __init__(self, tokeniser=None, lang_code=None):
-		self.tokenise = self._tokenise
-
-		if lang_code is None:
-			self.lang_code = DEFAULT_LANG_CODE
-			logger.info("No Nahuatl variant language code provided. "
-			            "Defaulting to `nhi`.")
-		else:
-			if lang_code not in SUPPORTED_LANG_CODES:
-				logger.error("Unsupported language variant specified.")
-				raise ValueError(f"Unsupported lang code for Nahuatl: "
-								 f"{lang_code}")
-			else:
-				self.lang_code = lang_code
-		if tokeniser:
-			self.tokenise = tokeniser
-
-		_path_to_att_dir = pkg_resources.files("elotl.nahuatl.data").joinpath(f"{self.lang_code}.mor.att")
-		_path_to_tsv_dir = pkg_resources.files("elotl.nahuatl.data").joinpath(f"{self.lang_code}.mor.tsv")
-
-		self.analyser = ATTFST(_path_to_att_dir)
-		self.convertor = elotl.utils.morphology.Convertor(_path_to_tsv_dir)
-		self.normaliser = Normaliser("ack")
+	def __init__(self, lang_code, tokenizer=None, normalize=True):
+		super().__init__(lang_code, tokenizer)
+		if normalize is True:
```

Comment:
Bastaría con algo como esto:

```suggestion
		if normalize:
```

Reply:
La idea aquí era evitar el caso de que se utiliza un tipo equivocado (como un str o func o algo)...

Reply:
de una vez trataré de agregar los type hints de todos modos y luego cambio esto